### DISCUSSION TITLE
cadquery OCCT , rust and stuff

### DISCUSSION THREAD

**Discussion by zignig** [bschwind/opencascade-rs#98] [Show and tell]
Been making 3d stuff for a while and have come across this project. 

3D , rust and an OCCT backend. Nice.

I have been working on a little side project that I will be using for my opencascade-rs development. 

https://github.com/zignig/stl_helper/

It watches folder for changed .stl files and then displays them in three.js in a browser.  It's nearly working.  just one more code change.... ;) 

I have done lots of cadquery / cqparts work some time ago and a rust library interested me because;  Rust ; and the whole single binary thing.

Carry On!!

zignig

**Comment by bschwind** [comment]
Great to hear you're interested!

The viewer app will also eventually support hot-reloading of models and hopefully much more after that.

I'm still experimenting with the best way to make it dynamic, and I'm currently leaning towards WASM for that but I need more experimentation to be sure.

Looking forward to see what you create with this library :)

**Comment by zignig** [comment]
for reference  my work with cadquery  and cqparts is https://cqparts.github.io/  , I drifted away from it some years ago. 

However , https://github.com/zignig/cqparts_bucket/ may give you some ideas on how to approach structured 3d things.

The translation between python and rust is a bit weird , but it should get your brain juices flowing ;)

**Comment by bschwind** [comment]
Thanks @zignig! I like the thoughts and concepts behind cqparts, especially going from shapes to parts, and then parts to compound objects with constraints. Seems quite modular and nice, I'll be keeping this design in mind when I move on to tackling part assemblies.

If I design things right, I'm hoping cargo will be a big boost to supporting user-created libraries that integrate well with this one so it becomes a breeze to pull in other parts to work off of.
